### DISCUSSION TITLE
Querystring and routing support

### DISCUSSION THREAD

**Discussion by gsxdsm** [tetra-framework/tetra#80] [Ideas]
In the tetra docs there was a mention of querystring and routing support. Would be good to discuss what this should look like and align on the API here. @samwillis @nerdoc

**Comment by nerdoc** [comment]
Sam added this to the initial goals. I have coded a bootstrap tab component that allows the user to pre-select a certain tab by using `?tab=xyz` in the url. this works well, but I have to think about how to implement this in a more generic way. The first version I did was with htmx and views - it's easy there to get the parameters.

However, in a component, you have direct access too to `request.GET`, so filtering out the e.g. "tab" query would be easy. It's just about how to design the API.

In my generic tab component, I had a "tab_query" attribute  where you could determine which GET variable was used for that component, default was "tab". If we implement this, there should be support for multiple query strings.

Best thing would be to collect use cases. Ideas?

* Generic "Tab" component, with `tab=foo` as GET param.
